.github/test_real.sh

```diff
@@ -1,3 +1,6 @@
 #!/bin/bash
 set -e
+# Set these to allow MPI oversubscription because the tests need to run on 3 procs but the test runner may only have 2
+export OMPI_MCA_rmaps_base_oversubscribe=1
+export PRTE_MCA_rmaps_default_mapping_policy=:oversubscribe
```

Comment:
Few comments:
* We should add comments for each line for OpenMPI 4 and 5
* We should only set this for the Azure runners
* This should not be set on intel images

Reply:
I think it's fine to set these all the time, not just when running on Azure, that way the tests will work for anyone running this script. How would you recommend checking for OpenMPI vs Intel MPI within the script?

Reply:
I pushed some changes to only set these on non-intel images.